pkg/extsvc/github/repos.go

```
@@ -379,6 +382,74 @@ query Repositories($ids: [ID!]!) {
 	return repos, nil
 }
 
+// GetRepositoriesByNameWithOwnerFromAPI fetches the specified repositories
+// (namesWithOwners) from the GitHub GraphQL API and returns a slice of
+// repositories.
+// If a repository is not found, it will return an error.
+//
+// The maximum number of repositories to be fetched is 30. If more
+// namesWithOwners are given, the method returns an error. 30 is not a official
+// limit of the API, but based on the observation that the GitHub GraphQL does
+// not return results when more than 37 aliases are specified in a query. 30 is
+// the conservative step back from 37.
+//
+// This method does not cache.
+func (c *Client) GetRepositoriesByNameWithOwnerFromAPI(ctx context.Context, token string, namesWithOwners []string) ([]*Repository, error) {
+	if len(namesWithOwners) > 30 {
+		return nil, errors.New("cannot fetch more than 30 repositories via GraphQL API")
+	}
+
+	query, err := c.buildGetRepositoriesBatchQuery(ctx, namesWithOwners)
+	if err != nil {
+		return nil, err
+	}
+
+	var result map[string]*Repository
+	err = c.requestGraphQL(ctx, token, query, map[string]interface{}{}, &result)
+	if err != nil {
+		if gqlErrs, ok := err.(graphqlErrors); ok {
```

This could be simplified like this:

```go
errs, _ := err.(graphqlErrors)
for _, e := range errs {
    if e.Type != graphqlErrTypeNotFound {
        return nil, e // I changed this from err too e. Did you mean to return the parent err?
    }
}
return nil, err
```

---

Thanks! I adopted the code from the other methods. Question is: do we want to return an error if _one_ repository was not found or not? I'd say it makes sense to ignore a missing repo, because we don't want the syncer to fail when someone deletes/renames a GitHub repository without telling the Sourcegraph admin before.